HLTrigger/special/plugins/HLTMuonRechitClusterFilter.cc

```diff
@@ -0,0 +1,130 @@
+  edm::InputTag cluster_tag_;
+  int min_N_;
+  int min_Size_;
+  int min_SizeMinusMB1_;
+  int max_nMB1_;
+  int max_nMB2_;
+  int max_nME11_;
+  int max_nME12_;
+  int max_nME41_;
+  int max_nME42_;
+  int min_nStation_;
+  double min_avgStation_;
+  double min_Time_;
+  double max_Time_;
+  double min_Eta_;
+  double max_Eta_;
+  double max_TimeSpread_;
+};
+//
+// constructors and destructor
+//
+HLTMuonRechitClusterFilter::HLTMuonRechitClusterFilter(const edm::ParameterSet& iConfig)
+    : HLTFilter(iConfig),
+      cluster_tag_(iConfig.getParameter<edm::InputTag>("ClusterTag")),
+      min_N_(iConfig.getParameter<int>("MinN")),
+      min_Size_(iConfig.getParameter<int>("Min_Size")),
+      min_SizeMinusMB1_(iConfig.getParameter<int>("Min_SizeMinusMB1")),
+      max_nMB1_(iConfig.getParameter<int>("Max_nMB1")),
+      max_nMB2_(iConfig.getParameter<int>("Max_nMB2")),
+      max_nME11_(iConfig.getParameter<int>("Max_nME11")),
+      max_nME12_(iConfig.getParameter<int>("Max_nME12")),
+      max_nME41_(iConfig.getParameter<int>("Max_nME41")),
+      max_nME42_(iConfig.getParameter<int>("Max_nME42")),
+      min_nStation_(iConfig.getParameter<int>("Min_nStation")),
+      min_avgStation_(iConfig.getParameter<double>("Min_avgStation")),
+      min_Time_(iConfig.getParameter<double>("Min_Time")),
+      max_Time_(iConfig.getParameter<double>("Max_Time")),
+      min_Eta_(iConfig.getParameter<double>("Min_Eta")),
+      max_Eta_(iConfig.getParameter<double>("Max_Eta")),
+      max_TimeSpread_(iConfig.getParameter<double>("Max_TimeSpread")) {
+  cluster_token_ = consumes<reco::MuonRecHitClusterCollection>(cluster_tag_);
+}
+
+HLTMuonRechitClusterFilter::~HLTMuonRechitClusterFilter() = default;
+
+void HLTMuonRechitClusterFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  makeHLTFilterDescription(desc);
+  desc.add<edm::InputTag>("ClusterTag", edm::InputTag("hltCSCrechitClusters"));
+  desc.add<int>("MinN", 1);
+  desc.add<int>("Min_Size", 50);
+  desc.add<int>("Min_SizeMinusMB1", 0);
+  desc.add<int>("Max_nMB1", 0);
+  desc.add<int>("Max_nMB2", 0);
+  desc.add<int>("Max_nME11", 0);
+  desc.add<int>("Max_nME12", 0);
+  desc.add<int>("Max_nME41", 0);
+  desc.add<int>("Max_nME42", 0);
+  desc.add<int>("Min_nStation", 0);
+  desc.add<double>("Min_avgStation", 0.0);
+  desc.add<double>("Min_Time", -999);
+  desc.add<double>("Max_Time", 999);
+  desc.add<double>("Min_Eta", -1.0);
+  desc.add<double>("Max_Eta", -1.0);
+  desc.add<double>("Max_TimeSpread", 999);
+  descriptions.add("hltMuonRechitClusterFilter", desc);
+}
+
+//
+// member functions
+//
+
+// ------------ method called on each new Event  ------------
+bool HLTMuonRechitClusterFilter::hltFilter(edm::Event& iEvent,
+                                           const edm::EventSetup& iSetup,
+                                           trigger::TriggerFilterObjectWithRefs& filterproduct) const {
```

Comment:
`filterproduct` is not being used at all. For the syntax of how to produce "TriggerObjects", you can follow one of the many examples in `HLTrigger/*/` (e.g. [this one](https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/HLTrigger/HLTfilters/plugins/HLTSinglet.cc#L116)). If producing "TriggerObjects" in the output file is not considered necessary, please convert this to a normal ([stream or global](https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkModuleTypes)) `EDFilter`.

Reply:
Actually, the data types we consider are of higher level (egamma, muons, jets, etc.), not rechits, so this plugin should indeed be and can only be a normal EDFilter...

Reply:
@Martin-Grunewald We would like to save the object passing this filter. This new object (MuonRechitCluster) is a cluster of rechits in the muon chambers, so it is a high-level object from the analysis point of view. It should be a fairly small object as you can see from the object definition. https://github.com/cms-sw/cmssw/blob/6dceacb8da167d7a05c0c3c7faddbc985b23b0b6/DataFormats/MuonReco/interface/MuonRecHitCluster.h  
Do we need to add a new Trigger object type in this case?

Reply:
Sorry, rechits are not high-level enough - for example, we also do not store Calo- or PF-rechits nor even PFclusters, only PFjets etc. It is not about the object size per se, we would only store 4-momentum anyway, but rather that we can not store everything so rechits are not considered.

Reply:
@Martin-Grunewald These are not rechits, they are clustered objects, so basically a jet in the muon chambers. Keeping only the 4-momentum is fine for us. This will be important for efficiency studies of the new trigger paths.

Reply:
Sorry - offline you redo this reconstruction anyway for analysis purposes. For the specific dedicated use case of trigger development, you'd run on the raw data format and rerun HLT anyway.